Dockerfile

```diff
@@ -1,9 +1,10 @@
 # Keep golang version aligned with latest yocto release
```

Comment:
We discussed this in [the past](https://github.com/mendersoftware/mender-artifact/pull/604/files#r1580639114). We cannot silently update the golang version.

However the intention now is to actually publish this image, so we cannot anymore lock to golang 1.17 here (obviously). So could you please extend your PR (scope creep coming!) with the following:
* Add a new Dockerfile (f.eks `Dockerfile.backcompat` or something like that) which still build the binary with golang 1.17 and keeps this comment
* Add a new CI job to just do `docker build` on this new Dockerfile

Reply:
> Having an extra Dockerfile for building for go1.17 seems a bit excessive. How about a simple gitlab job (`build:yoctocompat`) that runs `go build` in a `golang:1.17` container?

Actually I thought about that during this week of inactivity in this PR. It is not only better, but also aligned to other client repos when we had to do backwards compatibility jobs. Thanks